src/core/lib/security/authorization/cel_authorization_engine.cc

```diff
@@ -132,21 +132,20 @@ std::unique_ptr<mock_cel::Activation> CelAuthorizationEngine::CreateActivation(
       activation->InsertValue(kHeaders,
                               mock_cel::CelValue::CreateMap(headers_.get()));
     } else if (elem == kSourceAddress) {
-      absl::string_view source_address(args.GetPeerAddress());
+      std::string source_address(args.GetPeerAddress());
       if (!source_address.empty()) {
         activation->InsertValue(
-            kSourceAddress,
-            mock_cel::CelValue::CreateStringView(source_address));
+            kSourceAddress, mock_cel::CelValue::CreateString(&source_address));
```

Comment:
I don't think this change is correct.  It looks to me like the real `CelValue` implementation is storing only a `string_view` pointing to whatever string you pass in here, regardless of whether that string is passed in as a `string_view` or a `string`.  But this code is now constructing a local `string` variable and using it to construct the `CelValue`.  The local `string` variable will go out of scope as soon as this block ends, which will leave the `CelValue` pointing to a string that no longer exists.